docs/source/advanced_patterns.rst

```diff
@@ -0,0 +1,107 @@
+.. _advanced_communication_patterns:
+
+
+*******************************
+Advanced communication patterns
+*******************************
+
+Some advanced communication patterns are also implemented in osBrain. They are
+called channels, rather than sockets, as they are formed with multiple sockets.
+
+
+.. index:: async-req-rep
+.. _async_req_rep:
+
+Asynchronous Request-Reply channel
+==================================
+
+An asynchronous request-reply channel is very similar to a normal request-reply
+pattern, only in this case we want to process the reply asynchronously (i.e.:
+assign a handler for this reply and forget about it until it is received).
+
+This means that, when using an asynchronous request-reply channel, a handler
+must be specified not only for the replier but also for the requester.
+
+Example
+-------
+
+See the following example in which Bob sends an asynchronous request to Alice.
+Note that:
+
+- When Alice binds, she uses ``ASYNC_REP`` instead of ``REP``.
+- Bob is assigned a handler too to process Alice reply when it is received.
+- Bob sends the request and is automatically freed. It can log messages or do
+  other stuff.
+- When the reply is received, Bob automatically processes it.
+
+.. literalinclude:: ../../examples/async_req_rep.py
+
+That is the most basic example, but channels are a bit more flexible. When
+sending an asynchronous request we may want to manually specify a handler for
+the reply. This will overwrite the default handler specified when connecting to
+the ``ASYNC_REP`` server. Try to change the ``.send()`` call::
+
+   def deaf(agent, message):
+       agent.log_info('I am deaf...')
+
+
+   bob.send('alice', 'Hello, Alice!', handler=deaf)
+
+We can also specify a maximum wait time for the reply and some code to be
+executed in case the reply was not received after that wait. Try the example
```

Comment:
Maybe change "after that wait" to "after that wait time" or simply "after that time". It sounds nicer to me 😄